decisions/2020/06/25-continuous-app-delivery-discussion.md

```diff
@@ -0,0 +1,81 @@
+# 2020-06-25, Continuous App Delivery
+
+## Attending
+
+Austin, Erik, Lars, Stian, Zubair, Birk
+
+## Context
+
+The meeting was scheduled to identify and propose solutions for features that we need to support continuous app delivery for core applications.  
+
+Link to related issue and agenda: https://github.com/dhis2/notes/issues/114
+
+
+## Support multiple versions of the same app
+
+- This was top priority before the meeting, as we wanted to have admins test apps on the instance before installing a new versions.
+- People are accustomed to have a staging server, so the need for this is not as urgent.
+
+**Conclusion:**
+
+- This has been postponed for now as there are more urgent issues.
+- A nice to have down the line.
+
+
+## Support for overriding core apps by custom apps
+
+- To be able to update core apps we need them to install them as a "custom app".
+- The proposed solution was to have the server check for a custom app of the same name, and use that if it exists.
+- We need to support "old" urls (eg. "dhis-web-dashboard") for backwards compatability, these should redirect to a possibly new url (eg. "/apps")
+- The issue of collisions of app names were brought up, as users could unintentionally install apps by the same name of the core app, thus overriding it.
+- Installing apps should be a locked down permission, and admins should be aware of conflicts.
+- We could potentially notify the user the first time a "custom core app" is installed.
+
+**Conclusion:**
+
+- Going forward with implementation of overrides of core apps.
+- Jira issue for this: https://jira.dhis2.org/browse/DHIS2-4394
```

Comment:
The DHIS2-4394 is the Epic, the override task is this: https://jira.dhis2.org/browse/DHIS2-9092